AggregateMetrics/AggregateMetrics/AzureWebApp/RateCounter.cs

```diff
@@ -60,7 +60,7 @@ public MetricTelemetry GetValueAndReset()
                 return metric;
             }
 
-            metric.Value = (currentTime.Second - this.dateTime.Second != 0) ? ((double)this.lastValue - cacheHelper.GetCounterValue(this.name)) / (currentTime.Second - this.dateTime.Second) : 0;
+            metric.Value = (currentTime.Second - this.dateTime.Second != 0) ? (cacheHelper.GetCounterValue(this.name) - (double)this.lastValue) / (currentTime.Subtract(this.dateTime).Seconds) : 0;
```

Comment:
You should only calculate this `(currentTime.Second - this.dateTime.Second != 0)` once. There are cases when datetime calculated at different moments will be different due to reasons like computer time correction, different cache of different processors, etc.